src/data_structure/postprocess_results.jl

```diff
@@ -53,8 +53,11 @@ function _save_connection_avg_throughflow!(m::Model, key, connection_flow)
     sizehint!(avg_throughflow, length(connection()) * length(stochastic_scenario()) * length(time_slice(m)))
     for ((conn, n, d, s, t), value) in connection_flow
         start(t) >= m_start || continue
-        # NOTE: always assume that the flow goes from the first to the second node in `connection__from_node`
-        n_from, n_to, _other_nodes... = connection__from_node(connection=conn, direction=anything)
+        # NOTE: always assume that the flow goes from the first node in `connection__from_node` 
+        # to the first node in `connection__to_node` s.t. only one node for each direction is needed
+        # CAUTION: would cause trouble for multi-terminal connections (with >= 3 nodes linked)
+        n_from, n_from_others... = connection__from_node(connection=conn, direction=anything)
+        n_to, n_to_others... = connection__to_node(connection=conn, direction=anything)
```

Comment:
Thanks for this @nnhjy - although I think there are many other places in the code where we use this assumption. Changing it here and not everywhere else may lead to trouble. Could you please double check that the entire source code is consistent after this change?

Reply:
Ah sure, I will check it